pipeline/core/builder.py

```diff
@@ -127,6 +134,39 @@ def _convert_yaml_to_json(self, yaml_file_path: Path, output_path: Path) -> None
             logger.exception("Failed to convert %s to JSON", yaml_file_path)
             raise
 
+    def _rewrite_oss_links(self, content: str, target_language: str | None) -> str:
+        """Rewrite /oss/ links to include the target language.
+
+        Args:
+            content: The markdown content to process.
+            target_language: Target language ("python" or "js") or None to skip rewriting.
+
+        Returns:
+            Content with rewritten links.
+        """
+        if not target_language:
+            return content
+
+        def rewrite_link(match: re.Match) -> str:
+            """Rewrite a single link match."""
+            pre = match.group(1)  # Everything before the URL
+            url = match.group(2)  # The URL
+            post = match.group(3)  # Everything after the URL
+
+            # Only rewrite absolute /oss/ paths
+            if url.startswith("/oss/"):
+                parts = url.split("/")
+                # Insert full language name after "oss"
+                parts.insert(2, self.language_url_names[target_language])
```

Comment:
is this deducing the target language from the fenching? what if a link is outside of a fence?